### DISCUSSION TITLE
Potential bug in object key name

### DISCUSSION THREAD

**Discussion by Bin-security** [brexhq/substation#94] [General]
Hey @jshlbrd. I am writing to report a potential bug in the s3 sns handler. The function call at [line](https://github.com/brexhq/substation/blob/main/cmd/aws/lambda/substation/s3.go#L190) tries to download an object with a bucket name and object key. The object key in the SNS notification event is url encoded, according to [this document](https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-content-structure.html). We might need to url-decode the key first before downloading the object. Thanks.

**Comment by jshlbrd** [comment]
@Bin-security can you open an issue for this and include more detail about the deployment (S3 object names, Lambda configuration, etc.) and what errors you're getting? Thanks!
